crates/pallet-domains/src/block_tree.rs

```diff
@@ -211,18 +213,29 @@ pub(crate) fn verify_execution_receipt<T: Config>(
         return Err(Error::InvalidExecutionTrace);
     }
 
+    let maybe_domain_runtime_upgraded_at = {
+        let runtime_id = Pallet::<T>::runtime_id(domain_id).ok_or(Error::RuntimeNotFound)?;
+        DomainRuntimeUpgradeAt::<T>::get(runtime_id).remove(consensus_block_number)
```

Comment:
You are missing the storage update here. 

Also noticed that we do not check if the ER extrinsic root contains the domain set_code extrinsic is being set

Reply:
I see. okay